hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -590,6 +592,7 @@ namespace hazelcast {
                 }
 
                 activeConnections.put(response.member_uuid, connection);
+                active_connection_ids_.put(connection->getConnectionId(), connection);
```

Comment:
We should put to `active_connection_ids_` before we put to `activeConnections`. 
Although the window is small there is an ordering problem. An invocation can be done on a connection(we get the connection from activeConnections map), where it is not available on `active_connection_ids_` map.

Reply:
Good catch. fixed the order.